aten/src/ATen/native/native_functions.yaml

```diff
@@ -617,6 +617,17 @@
 - func: affine_grid_generator_backward(Tensor grad, int[] size, bool align_corners) -> Tensor
   variants: function
 
+- func: _is_all_true(Tensor self) -> Tensor
+  variants: function, method
+  dispatch:
+    CompositeExplicitAutograd: _is_all_true
+
+# Note: this function is only for testing.
+- func: _test_check_tensor(Tensor self) -> Tensor
```

Comment:
I added this private op for testing. Is that alright? I couldn't think of any other way to test `TORCH_CHECK_TENSOR` on batched tensors

Reply:
When I add these, I get these errors:

```
FAIL: test_register_functorch_batched_decomposition_[aten::_is_all_true] (__main__.TestFunctorchDispatcher)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/work2/kurtamohler/development/pytorch-1/torch/testing/_internal/common_utils.py", line 252, in instantiated_test
    test(self, **param_kwargs)
  File "/work2/kurtamohler/development/pytorch-1/torch/testing/_internal/common_utils.py", line 391, in test_wrapper
    return test(*args, **kwargs)
  File "/work2/kurtamohler/development/pytorch-1/test/functorch/test_vmap_registrations.py", line 391, in test_register_functorch_batched_decomposition
    assert registration in CompositeImplicitAutogradRegistrations, (
AssertionError: The registrations in BatchedDecompositions.cpp must be for CompositeImplicitAutograd operations. If your operation aten::_is_all_true is not CompositeImplicitAutograd, then please register it to the FuncTorchBatched key in another file.

======================================================================
FAIL: test_register_functorch_batched_decomposition_[aten::_test_check_tensor] (__main__.TestFunctorchDispatcher)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/work2/kurtamohler/development/pytorch-1/torch/testing/_internal/common_utils.py", line 252, in instantiated_test
    test(self, **param_kwargs)
  File "/work2/kurtamohler/development/pytorch-1/torch/testing/_internal/common_utils.py", line 391, in test_wrapper
    return test(*args, **kwargs)
  File "/work2/kurtamohler/development/pytorch-1/test/functorch/test_vmap_registrations.py", line 391, in test_register_functorch_batched_decomposition
    assert registration in CompositeImplicitAutogradRegistrations, (
AssertionError: The registrations in BatchedDecompositions.cpp must be for CompositeImplicitAutograd operations. If your operation aten::_test_check_tensor is not CompositeImplicitAutograd, then please register it to the FuncTorchBatched key in another file.
```

I'm not sure what to do. Do we not actually need this `OP_DECOMPOSE` since the functions are marked `CompositeExplicitAutograd`?

Reply:
`_test_check_tensor` should be CompositeImplicitAutograd, `_is_all_true` should be CompositeExplicitAutograd. The  OP_DECOMPOSE should be for `_test_check_tensor`. Does that configuration not work?

Reply:
Thanks, that works